samza-api/src/main/java/org/apache/samza/metrics/MetricsRegistry.java

```diff
@@ -64,6 +64,17 @@
    */
   <T> Gauge<T> newGauge(String group, Gauge<T> value);
 
+  /**
+   * Register an existing {@link org.apache.samza.metrics.Gauge}
+   * @param group Group for this Gauge
+   * @param value Initial value for the Gauge
+   * @param <T> Type the Gauge will be wrapping
+   * @param overrideExistingGauge Overwrite any existing gauges present for the same group and gauge
+   * @return Gauge was registered
+   */
+  <T> Gauge<T> newGauge(String group, Gauge<T> value, Boolean overrideExistingGauge);
```

Comment:
Are we defining a new method here with the additional parameter to avoid a large code change in adding a boolean to the definition above?

Wondering whether the two methods will coexist long term or if we see a way to unify eventually.

Reply:
Yes, making overriding to be the default behavior requires testing to ensure that it doesnt break existing metrics; I can work on that separately. 
Although I tried with a test-job and everything looked ok.

Reply:
If leaving them separate is what is needed to ensure the P2 timeline and we plan for the two living along side one another to be temporary, maybe we can open a ticket and indicate in a comment when / under what conditions clean up can happen?

Reply:
Done, added details in https://issues.apache.org/jira/browse/SAMZA-2417

Reply:
I'd prefer to not add a new variant here and update the existing behavior instead. It seems to fall into "fixing undefined behavior" territory, in that I don't think a reasonable user expectation would be to retain the old metric. We can fix this here and call it out in the "backwards incompatible / API changes" section of the PR description.

@cameronlee314 What do you think?

Reply:
I was debating doing that, but I'm not sure if it'd break some existing metrics-subsystem which relies on us retaining the first object rather than the last?

Reply:
@prateekm I rollbacked all most of the changes; we now just retain the newest metric. 
can you approve it if it looks ok, so we can merge.